### PR TITLE
get annotations within a certain time range

### PR DESCRIPTION
**Pull request by lucasgautheron**
- [x] tests
    - [x] `get_within_time_range`
    - [x] `get_segments_timestamps`
- [x] documentation
- [x] implementation
    - [x] `get_within_time_range`
    - [x] `get_segments_timestamps`
    - [ ] CLI

### CODE REVIEW

File: ChildProject/metrics.py [hunk 1]
```diff
@@ -244,7 +244,7 @@ def gamma(
         )
 
     dissim = CombinedCategoricalDissimilarity(
-        list(continuum.categories), delta_empty=1, alpha=alpha, beta=beta
+        delta_empty=1, alpha=alpha, beta=beta
```
Comment (alecristia) [inline ChildProject/metrics.py:247]: this is unrelated, right?

File: tests/test_annotations.py [hunk 2]
```diff
@@ -11,310 +11,462 @@
...
+    annotations = pd.read_csv("tests/data/time_range_annotations.csv")
+    matches = am.get_within_time_range(annotations, '09:00', '20:00')
+
+    truth = pd.read_csv('tests/truth/time_range.csv')
+
+    pd.testing.assert_frame_equal(
+        standardize_dataframe(matches, truth.columns),
+        standardize_dataframe(truth, truth.columns)
+    )
+
```
Comment (alecristia) [inline tests/test_annotations.py:269]: maybe i'm missing something, but do you have a check for the user inputting start/stop with wrong formatting? (eg 1pm or 1.13 etc)

**Review by alecristia — APPROVED** [review]

File: ChildProject/metrics.py (hunk 1 above)
Comment (lucasgautheron) [reply ChildProject/metrics.py:247]: yes; the tests were failing due to an upgrade of pygamma-agreement. had to change this for the tests to pass again

File: tests/test_annotations.py (hunk 2 above)
Comment (lucasgautheron) [reply tests/test_annotations.py:269]: Added (a7c60cb4c8b14b13221645c89709f96da329c75e)

File: ChildProject/annotations.py [hunk 3]
```diff
@@ -1034,8 +1036,19 @@ def get_within_time_range(
...
+            raise ValueError(
+                f"invalid value for start_time ('{start_time}'); should have HH:MM format instead"
+            )
+
+        try:
+            end_dt = datetime.datetime.strptime(end_time, "%H:%M")
+        except:
+            raise ValueError(
+                f"invalid value for end_time ('{end_time}'); should have HH:MM format instead"
+            )
```
Comment (alecristia) [inline ChildProject/annotations.py:1052]: we should foresee one more error category: start time is after end time

File: tests/test_annotations.py [hunk 4]
```diff
@@ -267,6 +267,14 @@ def test_within_time_range(project):
...
     )
 
+    exception_caught = False
+    try:
+       matches = am.get_within_time_range(annotations, '9am', '8pm')
+    except ValueError as e:
+        exception_caught = True
+
+    assert exception_caught, "no exception was thrown despite invalid times"
+
```
Comment (alecristia) [inline tests/test_annotations.py:277]: that's kind :)

**Review by alecristia — APPROVED** [review]

**Comment by lucasgautheron** [comment]
@alecristia, do you think any of these two functionalities are worth being implemented into the command-line interface?

**Review by alecristia — APPROVED** [review]
I don't think this needs to be in the CLI -- the python API is all we need!
